### PR TITLE
added VictoryPortal

### LINKED ISSUE #329: Investigate portal pattern for tooltips

### PR DESCRIPTION
**Pull request by nfcampos**
closes https://github.com/FormidableLabs/victory/issues/329

see issue for more detail

### CODE REVIEW

File: src/victory-container/victory-container.js [hunk 1]
```diff
@@ -79,6 +107,7 @@ export default class VictoryContainer extends React.Component {
         <title id="title">{this.props.title}</title>
         <desc id="desc">{this.props.desc}</desc>
         {this.props.children}
+        {React.cloneElement(this.props.portalComponent, {ref: "portal"})}
```
Comment (boygirl) [inline src/victory-container/victory-container.js]: I think string refs are slated for deprecation. https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute

Can you re-write this as a callback? Also, do you think there's a danger of naming collision with "portal"?

**Review by boygirl — CHANGES_REQUESTED** [review]
@nfcampos this all looks great. Approved pending my comment about refs.

File: src/victory-container/victory-container.js (hunk 1 above)
Comment (nfcampos) [reply src/victory-container/victory-container.js]: (i still end up always reaching for the string refs for some reason, will replace it)
naming collision with what?

Comment (boygirl) [reply src/victory-container/victory-container.js]: Never mind about the naming collision. My mistake.

Comment (nfcampos) [reply src/victory-container/victory-container.js]: The latest commit uses callback ref

**Review by boygirl — APPROVED** [review]

**Comment by boygirl** [comment]
Awesome! I'll merge this now. Thanks you!

**Pull request merged**